ksqldb-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -93,116 +94,97 @@ public void putSource(final DataSource dataSource, final boolean allowReplace) {
       });
     }
 
-    dataSources.put(dataSource.getName(), new SourceInfo(dataSource));
+    dataSources.put(dataSource.getName(),
+        (existing != null) ? existing.copyWith(dataSource) : new SourceInfo(dataSource));
```

Comment:
I'm a little confused here, shouldn't `existing` already have the `SourceName` as `dataSource`? why are we calling `copyWith(dataSource)`?

Reply:
Removed. We don't need to add the dataSource anymore when calling `putSource` with replace and if the dataSource already exists.

Reply:
I reverted back this change. I need to replace the existing dataSource because it may contain changes in the Schema (caused by AlterStream). But I need a copy of the internal linked/referential sources.